server/cache/dirtools/dirtools.go

```diff
@@ -420,10 +423,9 @@ func handleSymlink(dirHelper *DirHelper, rootDir string, cmd *repb.Command, acti
 		}
 		if symlinkInfo.IsDir() {
 			actionResult.OutputDirectorySymlinks = append(actionResult.OutputDirectorySymlinks, symlink)
-			return nil
+		} else {
+			actionResult.OutputFileSymlinks = append(actionResult.OutputFileSymlinks, symlink)
 		}
-
-		actionResult.OutputFileSymlinks = append(actionResult.OutputFileSymlinks, symlink)
```

Comment:
nit: we probably don't need this style change but I don't mind leaving it in

Reply:
It's not a functional change, but I think the symmetric form makes it more obvious that exactly one of these is set.